### PR TITLE
fix plugin window

### LINKED ISSUE #95: Plugins menu item is not accessible
`Settings` → `Plugins` menu item is not accessible.

The issue looks trivial though, as always, no one knows how far it extends.
```
Traceback (most recent call last):
  File "./gourmet/GourmetRecipeManager.py", line 1113, in <lambda>
    lambda *args: plugin_gui.show_plugin_chooser()),
  File "./gourmet/plugin_gui.py", line 154, in show_plugin_chooser
    pc = PluginChooser()
  File "./gourmet/plugin_gui.py", line 20, in __init__
    self.add_labels()
  File "./gourmet/plugin_gui.py", line 37, in add_labels
    self.window.vbox.pack_start(head,expand=False)
TypeError: Gtk.Box.pack_start() takes exactly 5 non-keyword arguments (2 given)
```

### PR DESCRIPTION
**Pull request by kirienko**
Fix "Plugins" menu item (#95):

1. The signature of `self.window.vbox` was changed — otherwise plugin window just didn't pop up
2. The `data_fun` formatting function was made a class function
3. `PolicyType` of `ScrolledWindow` was changed to `NEVER` — otherwise the plugin area (essentially `Gtk.Notebook`) had been rendered with the wrong height (that shouldn't be like this, but I don't know why it was happening)

Now the plugin window is working in the same manner as it used to be.

Closes: #95

### CODE REVIEW

File: gourmet/plugin_gui.py [hunk 1]
```diff
@@ -61,6 +61,16 @@ def make_list_store (self, plugin_list):
                 )
         return ls
 
+    @staticmethod
+    def data_fun(col, renderer, mod, itr, _):
```
Comment (maweki) [inline gourmet/plugin_gui.py]: if it's not too much of a hassle, could you annotate this with types? The ones you don't know or are not sure about you can annotate with `Any` and we can later make that an error in `mypy`.

Comment (maweki) [reply gourmet/plugin_gui.py]: But I am not sure about the extraction. Do we plan on using this function anywhere else?

File: gourmet/plugin_gui.py [hunk 2]
```diff
@@ -71,20 +81,15 @@ def make_treeview (self, plugin_list):
...
+        plugin_col.set_cell_data_func(text_renderer, self.data_fun)
         plugin_col.set_property('expand',True)
         plugin_col.set_property('min-width',250)
         tv.append_column(plugin_col)
         tv.append_column(active_col)
         ls = self.make_list_store(plugin_list)
         tv.set_model(ls)
-        sw = Gtk.ScrolledWindow(); sw.set_policy(Gtk.PolicyType.NEVER,Gtk.PolicyType.AUTOMATIC)
+        sw = Gtk.ScrolledWindow()
+        sw.set_policy(Gtk.PolicyType.NEVER, Gtk.PolicyType.NEVER)
```
Comment (maweki) [inline gourmet/plugin_gui.py:94]: Is this just whitespace? I know it's bad, but that can be done automatically later on

**Review by maweki — CHANGES_REQUESTED** [review]
I am not sure about the function extraction. Is this a function of concern for the whole class? I guess that's just taste. But since we touch the header anyway, we might as well add a type annotation.

Maybe we need a dedicated pack-start-pass through the code.

**Review by cydanil — CHANGES_REQUESTED** [review]
Generally good! I second the type hinting request, and I would ask to perhaps rename the function? `data_fun` is a bit silly.

What about `plugin_description_formatter`?

File: gourmet/plugin_gui.py (hunk 2 above)
Comment (kirienko) [reply gourmet/plugin_gui.py:94]: No, it wasn't just a whitespace, argument of `set_policy` was changed: `AUTOMATIC` → `NEVER`.

**Review by kirienko — COMMENTED** [review]
@cydanil agree! I changed that.

File: gourmet/plugin_gui.py (hunk 1 above)
Comment (kirienko) [reply gourmet/plugin_gui.py]: The function is just a bit oversized to be a lambda, and I extracted it for sake of potential future enhancements. (The resulting output doesn't look fancy now, IMO.)

I annotated it, please have a look.

**Review by cydanil — APPROVED** [review]

File: gourmet/plugin_gui.py [hunk 3]
```diff
@@ -61,6 +62,17 @@ def make_list_store (self, plugin_list):
                 )
         return ls
 
+    @staticmethod
+    def plugin_description_formatter(col: Gtk.TreeViewColumn, renderer: Gtk.CellRendererText,
+                                     mod: Gtk.ListStore, itr: Gtk.TreeIter, data: Optional[Any]) -> None:
```
Comment (maweki) [inline gourmet/plugin_gui.py]: `Optional[Any]` is redundant as `None` is already part of `Any`. Otherwise great

**Review by maweki — APPROVED** [review]
I am still not sure about the method extraction but otherwise I am fine with it :+1:

**Comment by kirienko** [comment]
Thanks for reviewing!

**Pull request merged**